lib/urllib.js

```diff
@@ -315,17 +315,31 @@ exports.requestWithCallback = function (url, args, callback) {
     // you can use this hook to change every thing.
     args.beforeRequest(options);
   }
-  var timer = null;
+  var connectTimer = null;
+  var responseTimer = null;
   var __err = null;
   var connected = false; // socket connected or not
   var keepAliveSocket = false; // request with keepalive socket
   var responseSize = 0;
   var responseAborted = false;
-  var done = function (err, data, res) {
-    if (timer) {
-      clearTimeout(timer);
-      timer = null;
+
+
+  var cancelConnectTimer = function() {
```

Comment:
都改了吧

Sent from my iPhone

> On Aug 5, 2016, at 7:28 PM, Yuwei Ba <notifications@github.com> wrote:
> 
> In lib/urllib.js:
> 
> >    var __err = null;
> >    var connected = false; // socket connected or not
> >    var keepAliveSocket = false; // request with keepalive socket
> >    var responseSize = 0;
> >    var responseAborted = false;
> > -  var done = function (err, data, res) {
> > -    if (timer) {
> > -      clearTimeout(timer);
> > -      timer = null;
> > +
> > +
> > +  var cancelConnectTimer = function() {
> 上面的 var done 要一起改了么
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>